### PR TITLE
FEATURE: 116 execute dataprocessor on result

### PR DESCRIPTION
**Pull request by DanielSiepmann**
Data processors can now be configured to be executed for each document while indexing AND for each search result after searching, before providing results to template.

Resolves: #116

### CODE REVIEW

File: Tests/Unit/Domain/Model/ResultItemTest.php [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    {
+        $originalData = [
+            'uid' => 10,
+            'title' => 'Some title',
+        ];
+        $expectedData = $originalData;
+
+        $subject = new ResultItem($originalData);
+        $this->assertSame(
+            'Some title',
```
Comment (justusmoroni) [inline Tests/Unit/Domain/Model/ResultItemTest.php]: Use `$originalData['title']` here.

File: Tests/Unit/Domain/Model/ResultItemTest.php [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    /**
+     * @test
+     */
+    public function plainDataCanBeRetrieved()
+    {
+        $originalData = [
+            'uid' => 10,
+            'title' => 'Some title',
+        ];
+        $expectedData = $originalData;
```
Comment (justusmoroni) [inline Tests/Unit/Domain/Model/ResultItemTest.php:37]: Can't you remove this  and just use `$originalData` in line 41?

File: Tests/Unit/Domain/Model/ResultItemTest.php [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    /**
+     * @test
+     */
+    public function existenceOfDataCanBeChecked()
+    {
+        $originalData = [
+            'uid' => 10,
+            'title' => 'Some title',
+        ];
+        $expectedData = $originalData;
```
Comment (justusmoroni) [inline Tests/Unit/Domain/Model/ResultItemTest.php]: Not used?

File: Tests/Unit/Domain/Model/ResultItemTest.php [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    /**
+     * @test
+     */
+    public function dataCanNotBeChanged()
+    {
+        $originalData = [
+            'uid' => 10,
+            'title' => 'Some title',
+        ];
+        $expectedData = $originalData;
```
Comment (justusmoroni) [inline Tests/Unit/Domain/Model/ResultItemTest.php]: Not used?

File: Tests/Unit/Domain/Model/ResultItemTest.php [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+    /**
+     * @test
+     */
+    public function dataCanNotBeRemoved()
+    {
+        $originalData = [
+            'uid' => 10,
+            'title' => 'Some title',
+        ];
+        $expectedData = $originalData;
```
Comment (justusmoroni) [inline Tests/Unit/Domain/Model/ResultItemTest.php]: Not used?

File: Tests/Unit/Domain/Model/SearchResultTest.php [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+                'title' => 'Some Title 3',
+            ],
+        ];
+
+        $subject = new SearchResult($originalSearchResultMock, $data);
+        $resultItems = $subject->getResults();
+
+        $this->assertCount(3, $resultItems);
+
+        $this->assertSame(10, $resultItems[0]['uid']);
```
Comment (justusmoroni) [inline Tests/Unit/Domain/Model/SearchResultTest.php]: Use `$data[0]['uid]` here?
And the same thing in the following 2 lines?

**Review by justusmoroni — CHANGES_REQUESTED** [review]

File: Tests/Unit/Domain/Model/ResultItemTest.php (hunk 2 above)
Comment (DanielSiepmann) [reply Tests/Unit/Domain/Model/ResultItemTest.php:37]: I could, but just to make sure that `$originalData` is not received as reference an modified, I use a copy.
This way I can make sure that I receive exactly the same as I passed into the constructor.

File: Tests/Unit/Domain/Model/ResultItemTest.php (hunk 1 above)
Comment (DanielSiepmann) [reply Tests/Unit/Domain/Model/ResultItemTest.php]: 👍

File: Tests/Unit/Domain/Model/ResultItemTest.php (hunk 3 above)
Comment (DanielSiepmann) [reply Tests/Unit/Domain/Model/ResultItemTest.php]: Yeah copy and paste... thanks.

File: Tests/Unit/Domain/Model/ResultItemTest.php (hunk 4 above)
Comment (DanielSiepmann) [reply Tests/Unit/Domain/Model/ResultItemTest.php]: Yeah copy and paste... thanks.

File: Tests/Unit/Domain/Model/ResultItemTest.php (hunk 5 above)
Comment (DanielSiepmann) [reply Tests/Unit/Domain/Model/ResultItemTest.php]: Yeah copy and paste... thanks.

File: Tests/Unit/Domain/Model/SearchResultTest.php (hunk 6 above)
Comment (DanielSiepmann) [reply Tests/Unit/Domain/Model/SearchResultTest.php]: 👍

**Review by justusmoroni — APPROVED** [review]

**Pull request merged**